### PR TITLE
update tests with qir in files and add missing gates to gateset 

### PR DESCRIPTION
**Pull request by cqc-melf**

### CODE REVIEW

File: pytket/qir/conversion/api.py [hunk 1]
```diff
@@ -50,9 +50,9 @@ def pytket_to_qir(
     :type name: str
     :param qir_format: format of the generated qir, default value is binary
     :type qir_format: QIRFormat
-    :param pyqir_0_7_compatibility: converts the output to be compatible with
+    :param pyqir_0_6_compatibility: converts the output to be compatible with
         pyqir 0.7, default value false
```
Comment (cqc-alec) [inline pytket/qir/conversion/api.py]:
```suggestion
        pyqir 0.6, default value false
```

File: pytket/qir/conversion/conversion.py [hunk 2]
```diff
@@ -323,7 +325,8 @@ def _rebase_command_to_gateset(self, command: Command) -> Optional[Circuit]:
         if optype not in self.module.gateset.base_gateset:
             circuit = Circuit(self.circuit.n_qubits, self.circuit.n_bits)
             circuit.add_gate(optype, params, args)
-            self.rebase_to_gateset.apply(circuit)
+            if self.rebase_to_gateset.apply(circuit):
+                raise ValueError("Gate not supported")
```
Comment (cqc-alec) [inline pytket/qir/conversion/conversion.py]: This is a bit weird, attempting a rebase pass and failing if it does anything. Can we just check that the circuit only contains supported gates?

File: pytket/qir/conversion/conversion.py [hunk 3]
```diff
@@ -341,7 +344,8 @@ def _rebase_op_to_gateset(self, op: OpType, args: List) -> Optional[Circuit]:
             params = op.params
             circuit = Circuit(self.circuit.n_qubits, self.circuit.n_bits)
             circuit.add_gate(optype, params, args)
-            self.rebase_to_gateset.apply(circuit)
+            if self.rebase_to_gateset.apply(circuit):
```
Comment (cqc-alec) [inline pytket/qir/conversion/conversion.py]: Same here.

File: tests/qir/test_pytket_qir.qir [hunk 4]
```diff
@@ -0,0 +1,40 @@
+; ModuleID = 'test_pytket_qir'
+source_filename = "test_pytket_qir"
```
Comment (cqc-alec) [inline tests/qir/test_pytket_qir.qir]: Can we rename all the `.qir` files to `.ll`? This is more usual I think.

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: pytket/qir/conversion/conversion.py (hunk 2 above)
Comment (cqc-melf) [reply pytket/qir/conversion/conversion.py]: Yes I can do that.

File: pytket/qir/conversion/conversion.py (hunk 3 above)
Comment (cqc-melf) [reply pytket/qir/conversion/conversion.py]: done in @cqc-melf
[add gateset predicate](https://github.com/CQCL/pytket-qir/pull/42/commits/833d1194d7fd7ffb2443ab8306137e62cb6d4f0e)
[833d119](https://github.com/CQCL/pytket-qir/pull/42/commits/833d1194d7fd7ffb2443ab8306137e62cb6d4f0e)

File: pytket/qir/conversion/conversion.py (hunk 2 above)
Comment (cqc-melf) [reply pytket/qir/conversion/conversion.py]: done in @cqc-melf
[add gateset predicate](https://github.com/CQCL/pytket-qir/pull/42/commits/833d1194d7fd7ffb2443ab8306137e62cb6d4f0e)
[833d119](https://github.com/CQCL/pytket-qir/pull/42/commits/833d1194d7fd7ffb2443ab8306137e62cb6d4f0e)

File: tests/qir/test_pytket_qir.qir (hunk 4 above)
Comment (cqc-melf) [reply tests/qir/test_pytket_qir.qir]: done in @cqc-melf
[rename .qir to .ll](https://github.com/CQCL/pytket-qir/pull/42/commits/e2ec29603f5665335294b18879538a0104bd3d82)
[e2ec296](https://github.com/CQCL/pytket-qir/pull/42/commits/e2ec29603f5665335294b18879538a0104bd3d82)

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**